x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/SecurityContext.java

```diff
@@ -150,6 +151,16 @@ public void executeAsUser(User user, Consumer<StoredContext> consumer, Version v
         }
     }
 
+    /**
+     * Compared to {@code ThreadContext#stashContext} this only clears the specific transient headers that are used to convey the
+     * authorization outcome, leaving the other headers in place, just like {@code ThreadContext#newStoredContext(true)}. When
+     * the returned {@code StoredContext} is closed, the original authorization transient headers are restored (if any did
+     * not originally exist, it will also not exist after the restore).
+     */
```

Comment:
It feels important to me to state clearly that other than the specified transient headers, nothing else gets restored. Any other new headers generated by the child actions will be kept.

Reply:
I went with Tim's suggestion from https://github.com/elastic/elasticsearch/pull/61621#discussion_r484152581, which I believe satisfies your requirement.